web3_proxy/src/frontend/authorization.rs

```diff
@@ -417,6 +425,11 @@ impl RequestMetadata {
             }
         }
 
+        // Fetch the balance from the cache
+        let latest_balance = match NonZeroU64::try_from(authorization.checks.user_id) {
+            Err(_) => Arc::new(AtomicF64::from(0.)),
+            Ok(x) => app.user_balance_cache.get(&x).unwrap_or_default(),
```

Comment:
I don't think `get` is the right thing here. if the cache entry gets evicted before this get is called, then a user with an actual balance might see `0` here. Maybe we want to call `balance_checks` here and use its return? Or maybe `get_or_insert_async` would work?

Reply:
ok I changed a bunch, specifically getting items from the cache.